File: CS/WindowsFormsApplication1/MyDockManager/MyDockPanel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraBars.Docking.Paint;
using DevExpress.XtraBars.Styles;
using DevExpress.Utils;
using DevExpress.XtraBars.Docking;
using DevExpress.XtraBars.Docking.Helpers;

namespace WindowsFormsApplication1
{
    [System.ComponentModel.DesignerCategory("")]
    public class MyDockPanel : DockPanel
    {
        public MyDockPanel()
        {
            DockLayout = new MyDockLayout(DockingStyle.Float, this);   
        }
        protected internal MyDockPanel(bool createControlContainer, DockingStyle dock, DockManager dockManager)
            : base(createControlContainer, dock, dockManager)
        {
            
        }

    }
}

File: VB/WindowsFormsApplication1/MyDockManager/MyDockPanel.vb
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.XtraBars.Docking
Imports DevExpress.XtraBars.Docking.Helpers

Namespace WindowsFormsApplication1

    <System.ComponentModel.DesignerCategory("")>
    Public Class MyDockPanel
        Inherits DockPanel

        Public Sub New()
            DockLayout = New MyDockLayout(DockingStyle.Float, Me)
        End Sub

        Protected Friend Sub New(ByVal createControlContainer As Boolean, ByVal dock As DockingStyle, ByVal dockManager As DockManager)
            MyBase.New(createControlContainer, dock, dockManager)
        End Sub
    End Class
End Namespace
